src/main/java/frc/robot/commands/scoring/intake/IntakeGamePiece.java

```diff
@@ -2,34 +2,48 @@
 
 import edu.wpi.first.wpilibj2.command.CommandBase;
 import frc.robot.constants.IntakeConstants;
-import frc.robot.subsystems.Intake;
+import frc.robot.subsystems.RollerIntake;
+import frc.robot.subsystems.RollerIntake.IntakeMode;
+import frc.robot.subsystems.RollerIntake.IntakePiece;
 
 public class IntakeGamePiece extends CommandBase {
-  Intake m_intake; 
+
+  private final RollerIntake m_intake; 
+  private final IntakePiece m_type;
 
   /**
    * Spins the intake until the game piece is inside the intake.
    * @param intake the intake subsystem
    */
-  public IntakeGamePiece(Intake intake){
+  public IntakeGamePiece(RollerIntake intake, IntakePiece type) {
     m_intake = intake; 
+    m_type = type;
     addRequirements(m_intake);
   }
 
   @Override
   public void initialize() {
-    m_intake.intake(IntakeConstants.kIntakePower);
+    if (m_type == IntakePiece.CUBE) {
+      m_intake.setIntakeMode(IntakeMode.INTAKE_CUBE);
+    } else if (m_type == IntakePiece.CONE) {
+      m_intake.setIntakeMode(IntakeMode.INTAKE_CONE);
+    }
   }
 
   @Override
   public void end(boolean interrupted) {
-    m_intake.stopIntake(); 
+    if (!interrupted) m_intake.setHeldGamePiece(m_type);
```

Comment:
this should probably happen even if it is interrupted, because the driver can interrupt by releasing the controller earlier.
Even if it is interrupted and does set an that it has a game piece, it wouldn't matter because the robot would have to intake another game piece, setting it to the right mode, before outtaking.